app/code/community/Inchoo/SocialConnect/Block/Facebook/Button.php

```diff
@@ -35,16 +40,24 @@ protected function _construct() {
 
         $this->userInfo = Mage::registry('inchoo_socialconnect_facebook_userinfo');
 
-        // CSRF protection
-        Mage::getSingleton('core/session')->setFacebookCsrf($csrf = md5(uniqid(rand(), true)));
-        $this->client->setState($csrf);
-
         Mage::getSingleton('customer/session')
             ->setSocialConnectRedirect(Mage::helper('core/url')->getCurrentUrl());
 
         $this->setTemplate('inchoo/socialconnect/facebook/button.phtml');
     }
 
+    protected function _beforeToHtml()
+    {
+        if (!static::$csrf) {
+            static::$csrf = Mage::getSingleton('core/session')->getFacebookCsrf();
```

Comment:
Err... `facebook_csrf` is not set on the session yet. Shouldn't that be `md5(uniqid(rand(), true))`?

Reply:
🤦‍♂